### DISCUSSION TITLE
MacOS suddenly propmting "An update is ready to install" for brew-installed apps

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [Homebrew/discussions#5098] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: cfac516ee045d869be34598152e80a11ef2befe7
Last commit: 2 days ago
Core tap JSON: 31 Jan 00:07 UTC
Core cask tap JSON: 31 Jan 00:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.1xtpAb61Xi/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I don't recall this being the case before, but I notice now that when I start up my computer I am getting GUI pop-up prompts asking me to enter my password to update things like Visual Studio Code (i.e `"Code: An update is ready to install. Code is trying to add a new helper tool.`), when both were installed using brew on a dedicated admin account as the FAQ suggests for multi-user setups.

Is this a new macOS update? Has it always been this way? I assumed installing an app with brew would make apps defer to brew for all upgrading. Is there a way to force these popups to stop? Some apps I have noticed will repeatedly prompt me every 10 seconds even if I click decline, forcing me to quit the application. I do not want to update these apps via the GUI as I am worried it will mess with their brew-oriented installation and management.

**Comment by SMillerDev** [comment]
> Code is trying to add a new helper tool.

This sounds like it's coming from Code

> I assumed installing an app with brew would make apps defer to brew for all upgrading.

Generally only commandline apps do

> I do not want to update these apps via the GUI as I am worried it will mess with their brew-oriented installation and management.

Brew knows which apps do this and defer to those apps for updates unless you run `brew upgrade` with `--greedy`.
